### ISSUE TITLE
enhancement: provide additional resource snap points for action card preludes

### ISSUE THREAD

**Issue by fallspectrum** [Quinnsicle/arcs_tts#24]
This adds tags (Action + Resource) to the pre-existing snap points.  And adds 3 spread-out snap points (only for resources) on each action card spot (total of 12 new snap points).  Doing so makes it easier for players to "splay" out their prelude resources so it's easier to follow all the prelude actions they'are taking.  Compared to stacked resources which can make it non-obvious how the player is performing a combo.  These new resource snap points are still cleaned up by end_round.

![action_resource_snap_points_max](https://github.com/user-attachments/assets/f6ee3a6c-52c1-4324-a6be-7b6a357330cf)

Remove the following snap points:

```
{
      "Position": {
        "x": -11.2023087,
        "y": 0.9868063,
        "z": -1.31991363
      },
      "Rotation": {
        "x": -0.0008270628,
        "y": 89.99774,
        "z": -0.000334514072
      }
    },
    {
      "Position": {
        "x": -12.1977558,
        "y": 0.9867914,
        "z": -1.627565
      },
      "Rotation": {
        "x": -0.000827067764,
        "y": 89.99653,
        "z": -0.000334508833
      }
    },
    {
      "Position": {
        "x": -12.1999989,
        "y": 0.9868044,
        "z": 0.9199998
      },
      "Rotation": {
        "x": -0.0008270146,
        "y": 89.99754,
        "z": -0.000334644632
      }
    },
    {
      "Position": {
        "x": -12.1999989,
        "y": 0.9868164,
        "z": 3.479997
      },
      "Rotation": {
        "x": -0.000827127253,
        "y": 89.99208,
        "z": -0.000334461336
      }
    },
    {
      "Position": {
        "x": -11.1999989,
        "y": 0.986817062,
        "z": 1.24000025
      },
      "Rotation": {
        "x": -0.000827005832,
        "y": 89.9976,
        "z": -0.000334612065
      }
    },
    {
      "Position": {
        "x": -11.1999989,
        "y": 0.9868321,
        "z": 3.79999948
      },
      "Rotation": {
        "x": -0.00082710397,
        "y": 89.9920959,
        "z": -0.000334469049
      }
    },
```

```
{
      "Position": {
        "x": -12.198513,
        "y": 1.01333725,
        "z": -4.733032
      },
      "Rotation": {
        "x": -0.000874067366,
        "y": 90.0138,
        "z": 179.999741
      }
    },
    {
      "Position": {
        "x": -11.2030869,
        "y": 1.01336932,
        "z": -4.42531633
      },
      "Rotation": {
        "x": -0.000503447955,
        "y": 90.01388,
        "z": 180.000015
      }
    },
```

Then add the following.  Requires that "Action" tag is defined on all Action Cards + Deck objects that contain action cards.

```
    {
      "Position": {
        "x": -11.1948757,
        "y": 1.01336384,
        "z": -4.3894906
      },
      "Rotation": {
        "x": 0.0281503145,
        "y": 89.99729,
        "z": 180.020081
      },
      "Tags": [
        "Action",
        "Resource"
      ]
    },
    {
      "Position": {
        "x": -12.1984358,
        "y": 1.01333582,
        "z": -4.733103
      },
      "Rotation": {
        "x": 0.028526146,
        "y": 90.00026,
        "z": 0.0174652133
      },
      "Tags": [
        "Action",
        "Resource"
      ]
    },
    {
      "Position": {
        "x": -11.20218,
        "y": 1.0133667,
        "z": -1.3199054
      },
      "Rotation": {
        "x": 0.00539923226,
        "y": 89.9953,
        "z": 180.000916
      },
      "Tags": [
        "Action",
        "Resource"
      ]
    },
    {
      "Position": {
        "x": -12.1977491,
        "y": 1.01335454,
        "z": -1.62756681
      },
      "Rotation": {
        "x": -0.0005811547,
        "y": 90.00013,
        "z": -0.00247694273
      },
      "Tags": [
        "Action",
        "Resource"
      ]
    },
    {
      "Position": {
        "x": -11.1998587,
        "y": 1.013404,
        "z": 1.24000323
      },
      "Rotation": {
        "x": 0.005428973,
        "y": 89.99569,
        "z": 180.000946
      },
      "Tags": [
        "Action",
        "Resource"
      ]
    },
    {
      "Position": {
        "x": -12.1999722,
        "y": 1.01336694,
        "z": 0.9199967
      },
      "Rotation": {
        "x": -0.000575713057,
        "y": 90.0006,
        "z": -0.00251570786
      },
      "Tags": [
        "Action",
        "Resource"
      ]
    },
    {
      "Position": {
        "x": -11.1998682,
        "y": 1.01341629,
        "z": 3.800008
      },
      "Rotation": {
        "x": 0.005429279,
        "y": 89.99544,
        "z": 180.000946
      },
      "Tags": [
        "Action",
        "Resource"
      ]
    },
    {
      "Position": {
        "x": -12.1999779,
        "y": 1.01338589,
        "z": 3.47999287
      },
      "Rotation": {
        "x": -0.0005755585,
        "y": 90.00059,
        "z": -0.00251577864
      },
      "Tags": [
        "Action",
        "Resource"
      ]
    },
    {
      "Position": {
        "x": -10.069149,
        "y": 1.01338315,
        "z": -4.857407
      },
      "Rotation": {
        "x": 0.10576956,
        "y": 89.86849,
        "z": 0.00216878881
      },
      "Tags": [
        "Resource"
      ]
    },
    {
      "Position": {
        "x": -13.2026882,
        "y": 1.01333392,
        "z": -4.13493538
      },
      "Rotation": {
        "x": 358.742249,
        "y": 90.1329,
        "z": 0.12676321
      },
      "Tags": [
        "Resource"
      ]
    },
    {
      "Position": {
        "x": -10.1215515,
        "y": 1.0134002,
        "z": -1.84032917
      },
      "Rotation": {
        "x": 0.178341091,
        "y": 90.050766,
        "z": 0.01660332
      },
      "Tags": [
        "Resource"
      ]
    },
    {
      "Position": {
        "x": -13.2872334,
        "y": 1.01335251,
        "z": -1.05237913
      },
      "Rotation": {
        "x": 1.554699,
        "y": 270.155365,
        "z": 0.068159014
      },
      "Tags": [
        "Resource"
      ]
    },
    {
      "Position": {
        "x": -10.0954762,
        "y": 1.01341462,
        "z": 0.7468088
      },
      "Rotation": {
        "x": 359.862335,
        "y": 269.4909,
        "z": 0.00290514715
      },
      "Tags": [
        "Resource"
      ]
    },
    {
      "Position": {
        "x": -13.2279911,
        "y": 1.01336849,
        "z": 1.55490649
      },
      "Rotation": {
        "x": 358.514648,
        "y": 89.98057,
        "z": 0.0388385132
      },
      "Tags": [
        "Resource"
      ]
    },
    {
      "Position": {
        "x": -10.189805,
        "y": 1.013431,
        "z": 3.158328
      },
      "Rotation": {
        "x": 1.82725465,
        "y": 89.4129257,
        "z": 358.7991
      },
      "Tags": [
        "Resource"
      ]
    },
    {
      "Position": {
        "x": -13.2274323,
        "y": 1.01338708,
        "z": 4.167263
      },
      "Rotation": {
        "x": 358.499969,
        "y": 90.07014,
        "z": 359.988922
      },
      "Tags": [
        "Resource"
      ]
    },
    {
      "Position": {
        "x": -10.1749353,
        "y": 1.01338673,
        "z": -3.67388225
      },
      "Rotation": {
        "x": 0.266566873,
        "y": 89.5206146,
        "z": 0.105302945
      },
      "Tags": [
        "Resource"
      ]
    },
    {
      "Position": {
        "x": -10.159421,
        "y": 1.01340592,
        "z": -0.6706625
      },
      "Rotation": {
        "x": 0.151146218,
        "y": 89.61195,
        "z": 0.208042473
      },
      "Tags": [
        "Resource"
      ]
    },
    {
      "Position": {
        "x": -10.1773949,
        "y": 1.01342344,
        "z": 1.89964664
      },
      "Rotation": {
        "x": 0.10755121,
        "y": 89.80822,
        "z": 5.85254335
      },
      "Tags": [
        "Resource"
      ]
    },
    {
      "Position": {
        "x": -10.10196,
        "y": 1.01344144,
        "z": 4.32735252
      },
      "Rotation": {
        "x": 0.2298898,
        "y": 89.48307,
        "z": 0.145286217
      },
      "Tags": [
        "Resource"
      ]
    }
```

**Comment by fallspectrum** [comment]
Adding the Action tags dynamically may be required as I'm now noticing: https://github.com/Quinnsicle/arcs_tts/blob/main/src/ActionCards.lua#L72-L83

The following seems to address that:
```
function ActionCards.setup_deck(player_count)
    local four_player_deck = getObjectFromGUID(action_deck_4P_GUID)
    local deck = ActionCards.get_action_deck()
    if (player_count == 4) then
        deck.putObject(four_player_deck)
        Wait.time(function()
            deck.randomize()
            -- Add "Action" tag to all cards in the deck
            for _, card in ipairs(deck.getObjects()) do
                deck.addTag("Action", card.guid)
            end
        end, 1.5)
    else
        destroyObject(four_player_deck)
    end
end

```

**Comment by fallspectrum** [comment]
Resolved by #31
